Point_set_processing_3/include/CGAL/Point_with_normal_3.h

```diff
@@ -147,7 +147,8 @@ struct Normal_of_point_with_normal_pmap
   typedef Normal_of_point_with_normal_pmap<Gt> Self;
   /// \name Put/get free functions
   /// @{
-  friend reference get(const Self&,const key_type& k) {return k.normal();}
+  friend const reference get(const Self&,const key_type& k) {return k.normal();}
```

Comment:
```C++
const key_type key;
typedef typename boost::property_traits<Pmap>::reference reference;
reference v = get(pmap, key);
```

Unless I am completely wrong, that code will not compile because `reference` is a non-const `value_type&`, but the call to `get(pmap, key)` with the `const` key will return a const reference `const value_type&`...

**actually no!** `const reference` is not `const value_type&`! A reference is always const because its value (as a pointer) cannot be changed. `const reference` is the same type as `reference`. That code should not compile, because, with `const key_type& k`, `k.normal()` should not bind to a reference to a non-const value.

Try this code:
```C++
int main() {
  typedef int& reference;
  int i = 0;
  const reference r = i;
  r = 1;
  return r;
}
```